### PR TITLE
Xqueuejob client code part 1

### PR DESCRIPTION
**Pull request by dmatch01**
Part 1 only includes new files.  This is the first part of a xlarge PR for enhanced queuejob.

### CODE REVIEW

**Comment by scostache** [comment]
How is this different from my previous PR: 253 ?

**Comment by dmatch01** [comment]
@scostache This is a subset of 253.  I was starting to break up 253 such that it would not be in XXL category.  This was in response to @k82cn request to make the PRs smaller.  

@k82cn Is this PR size workable for you?  Please advise.

File: deployment/Dockerfile.queue-ctrl [hunk 1]
```diff
@@ -0,0 +1,3 @@
+From ubuntu:18.04
```
Comment (k82cn) [inline deployment/Dockerfile.queue-ctrl:1]: why we need this dockerfile?

File: pkg/controller/xqueuejob/queuejob_controller_ex.go [hunk 2]
```diff
@@ -0,0 +1,240 @@
+/*
```
Comment (k82cn) [inline pkg/controller/xqueuejob/queuejob_controller_ex.go:1]: please remove the controller in this pr.

Comment (scostache) [reply pkg/controller/xqueuejob/queuejob_controller_ex.go:1]: @k82cn : see PR#253

Comment (k82cn) [reply pkg/controller/xqueuejob/queuejob_controller_ex.go:1]: Would you sync up before open PRs ?

**Pull request closed without merging**